runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/resources/PipelineResources.java

```diff
@@ -49,7 +50,19 @@
       ClassLoader classLoader, PipelineOptions options) {
 
     PipelineResourcesOptions artifactsRelatedOptions = options.as(PipelineResourcesOptions.class);
-    return artifactsRelatedOptions.getPipelineResourcesDetector().detect(classLoader);
+    return artifactsRelatedOptions
+        .getPipelineResourcesDetector()
+        .detect(classLoader)
+        .filter(isStageable())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Returns a predicate for filtering all resources that are impossible to stage (like gradle
+   * wrapper jars).
+   */
+  private static Predicate<String> isStageable() {
+    return resourcePath -> !resourcePath.contains("gradle/wrapper");
```

Comment:
Is there a more programmatic way to filter these out, e.g. to stop at the Gradle wrapper's classloader?

Reply:
I tried to use classgraph's `blacklist*()` methods to filter these. However, I couldn't get any of them working for this particular directory. 

I still think that even despite that, filtering should also be possible (and easy, that's why I used streams) outside detector's implementation.

Reply:
Let me investigate this a little bit more.